File: LeetCode/116. 填充每个节点的下一个右侧节点指针.ts
/**
 * Definition for Node.
 * class Node {
 *     val: number
 *     left: Node | null
 *     right: Node | null
 *     next: Node | null
 *     constructor(val?: number, left?: Node, right?: Node, next?: Node) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function connect(root: Node | null): Node | null {
    if (root === null) return null
    const connect2Nodes = (a: Node | null,b: Node | null) => {
        if (a === null || b === null) return
        a.next = b
        connect2Nodes(a.left, a.right)
        connect2Nodes(a.right, b.left)
        connect2Nodes(b.left, b.right)
    }
    connect2Nodes(root.left, root.right)
    return root 
};

File: LeetCode/116. 填充每个节点的下一个右侧节点指针.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val: int = 0, left: 'Node' = None, right: 'Node' = None, next: 'Node' = None):
        self.val = val
        self.left = left
        self.right = right
        self.next = next
"""

class Solution:
    def connect(self, root: 'Optional[Node]') -> 'Optional[Node]':
        def connectTwo(pre: 'Optional[Node]',next: 'Optional[Node]'):
            if pre and next:
                pre.next = next;
                connectTwo(pre.left, pre.right)
                connectTwo(pre.right, next.left)
                connectTwo(next.left, next.right)
        if root:
            connectTwo(root.left, root.right)
        return root
